src/server_manager/web_app/ui_components/outline-region-picker-step.ts

```diff
@@ -127,33 +131,26 @@ export class OutlineRegionPicker extends LitElement {
           </label>`;
         })}
       </div>
-      ${this.isServerBeingCreated ? html`<paper-progress indeterminate="" class="slow"></paper-progress>` : ''}
+      ${
+        this.isServerBeingCreated ?
+            html`<paper-progress indeterminate="" class="slow"></paper-progress>` :
+            ''}
     </outline-step-view>
     `;
   }
 
   reset(): void {
-    this.isServerBeingCreated = false;
     this.selectedLocationId = null;
   }
 
-  _isCreateButtonEnabled(isCreatingServer: boolean, selectedLocationId: string): boolean {
-    return !isCreatingServer && selectedLocationId != null;
-  }
-
   _locationSelected(event: Event): void {
     const inputEl = event.target as HTMLInputElement;
     this.selectedLocationId = inputEl.value;
   }
 
   _handleCreateServerTap(): void {
-    this.isServerBeingCreated = true;
-    const params = {
-      bubbles: true,
-      composed: true,
-      detail: {selectedRegionId: this.selectedLocationId}
-    };
-    const customEvent = new CustomEvent('RegionSelected', params);
+    const params = {bubbles: true, composed: true, detail: {regionId: this.selectedLocationId}};
```

Comment:
Perhaps we don't need those parameters, since you are adding the event listener to the element directly.
Could you try using an Event without those parameters?